Geometry/CaloGeometry/interface/FlatTrd.h

```diff
@@ -71,13 +71,16 @@ class FlatTrd : public CaloCellGeometry {
   
   void getTransform( Tr3D& tr, Pt3DVec* lptr ) const override;
 
+  void setPosition ( const GlobalPoint& p ) { m_global = p;  setRefPoint(p); }
+  void setBackPoint( const GlobalPoint& p ) { setBackPoint(p); }
```

Comment:
This is wrong. This will result in infinite recursion until the stack is exhausted.